test/CadlRanchProjects/client/structure/multi-client/src/Generated/ClientAClient.cs

```diff
@@ -34,23 +35,20 @@ protected ClientAClient()
 
         /// <summary> Initializes a new instance of ClientAClient. </summary>
         /// <param name="endpoint"> Need to be set as 'http://localhost:3000' in client. </param>
-        /// <param name="client"> Need to be set as 'default', 'multi-client', 'renamed-operation', 'two-operation-group' in client. Allowed values: "default" | "multi-client" | "renamed-operation" | "two-operation-group". </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="client"/> is null. </exception>
-        /// <exception cref="ArgumentException"> <paramref name="client"/> is an empty string, and was expected to be non-empty. </exception>
-        public ClientAClient(Uri endpoint, string client) : this(endpoint, client, new ClientStructureServiceMultipleClientOptions())
+        /// <param name="client"> Need to be set as 'default', 'multi-client', 'renamed-operation', 'two-operation-group' in client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> is null. </exception>
+        public ClientAClient(Uri endpoint, ClientType client) : this(endpoint, client, new ClientStructureServiceMultipleClientOptions())
```

Comment:
considering the fact that the client actually does not have the concept of protocol or convenience, is it OK that the client accepts an enum?
I believe we generate this as a string on purpose.
I think this change we should go over with Michael.